extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -538,14 +538,21 @@ export async function promptConnectionStringPasswordAndUpdateConnectionString(co
 				connectionString = await vscodeMssqlApi.getConnectionString(connectionDetails, true, false);
 			}
 		}
-		// set connection string to not include the password if connection info does not include password, or user chooses to not include password, or authentication type is not sql login
-		let userPassword: string | undefined;
-		if (includePassword !== constants.yesString) {
+		// set connection string to not include the password if connection info does not include password, 
+		// or user chooses to not include password, or authentication type is not SQL login
+		if (includePassword !== constants.yesString || !connectionInfo.password) {
 			connectionString = await vscodeMssqlApi.getConnectionString(connectionDetails, false, false);
-
-			// Ask user to enter password if auth type is sql login and password is not saved
-			if (connectionInfo.authenticationType === 'SqlLogin' && connectionInfo.password) {
-				userPassword = await vscode.window.showInputBox({
+			// show warning message that user will have to enter it manually later in local.settings.json if they choose to 
+			// not to include password or if connection info does not include password
+			void vscode.window.showWarningMessage(constants.userPasswordLater, constants.openFile, constants.closeButton).then(async (result) => {
```

Comment:
So we both show them the warning message and then always prompt them for a password? That still doesn't seem correct. 

Here's the scenarios and how I currently believe they should be handled (SQL auth only, non-password connections skip this entirely)

1. Connection had password saved and user chose to include it : Connection string is saved without password, warning NOT shown and user NOT prompted to enter manually
2. Connection had password saved and user chose NOT to include it : Connection string is saved without password, warning shown and user NOT prompted to enter manually
3. Connect did NOT have password saved : User is asked whether they want to enter password and save in local.settings.json
    * If they select yes then get password and write to settings, warning NOT shown
    * If they select no then display warning

Reply:
So in scenario 3 instead of asking for another prompt to ask to enter a password or not. 

I can show the user to manually enter it and if they cancel then we show them the warning message.

Reply:
As long as we make it explicitly clear that when they enter the password it's going to be saved in the local.settings.json file. That's why I added the extra prompt. But if you can think of a better way that's fine.

Reply:
Yeah I think having a single prompt asks the user to enter the password to add it local.setting.json file and escape to cancel would suffice in this regard

Reply:
<img width="504" alt="image" src="https://user-images.githubusercontent.com/23587151/168931911-02b27f32-1172-4ab7-8389-237415386839.png">